modules/fusioncatcher/fusioncatcher/main.nf

```diff
@@ -0,0 +1,64 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process FUSIONCATCHER {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::fusioncatcher=1.33" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/fusioncatcher:1.33--hdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/fusioncatcher:1.33--hdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    path resource
+
+    output:
+    tuple val(meta), path("*.fusioncatcher-fusion-genes.txt")   , optional:true, emit: fusions
+    tuple val(meta), path("*.fusioncatcher_summary.txt")        , optional:true, emit: summary
+    tuple val(meta), path("*.fusioncatcher.log")                               , emit: log
+    path "*.version.txt"                                                       , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def fastq    = reads.size() == 2 ? "${reads[0]},${reads[1]}" : "${reads[0]}"
+    def valid    = (resource.exists() && file("${resource}/organism.txt").exists()) ? true : false
+
+    if (valid) {
+
+        """
+        fusioncatcher \\
+            -d $resource \\
+            -i $fastq \\
+            -p $task.cpus \\
+            -o . \\
+            --aligners=star
+            --skip-blat \\
+            $options.args
+
+        mv final-list_candidate-fusion-genes.txt ${prefix}.fusioncatcher-fusion-genes.txt
+        mv summary_candidate_fusions.txt ${prefix}.fusioncatcher_summary.txt
+        mv fusioncatcher.log ${prefix}.fusioncatcher.log
+
+        fusioncatcher --version | sed 's/fusioncatcher.py //' > ${software}.version.txt
+        """
+
+    }
+    else{
+
+        """
+        echo "Error: Invalid genome database" > ${prefix}.fusioncatcher.log
```

Comment:
Wouldn't you want the process to fail instead of just silently writing something to the log message?

Reply:
@grst Thanks for the comment. I know it is not the best way, but I did it just to pass the CI test as we don't the genome database, which is pretty large to use for the test. Please suggest what is the appropriate way to handle it.

Reply:
It would be ideal to generate a small reference genome database based on the reduced test-genomes in the `test-datasets` repository. It is also possible to generate a reference genome on-the-fly from those test data if you have a separate module to generate the genome database. 

If this doesn't work on a toy dataset, we need to think of a different solution.

Reply:
The files are provided directly to download, unzip and use / or built one using the build script for organisms in Ensembl database. It creates data (directory) of size 22GB for human. We cannot limit to specific chromosome. The building step will take ~5hrs, according to https://github.com/ndaniel/fusioncatcher/blob/master/doc/manual.md#5---quick-installation. I don't think we can create one for the other test-genome in the repo. e.g. sarscov2

Reply:
I'm thinking that this process is very specific for the rnafusion pipeline, and we can't really CI it, so maybe just have it as a local module for the rnafusion pipeline?

Reply:
@maxulysse - maybe yes for now. Shall I go with it as local module?

Reply:
@grst what do you think?

Reply:
local module sounds good for now

Reply:
Ok! I shall close this PR.